proton-j/src/main/java/org/apache/qpid/proton/engine/impl/EventImpl.java

```diff
@@ -68,12 +69,19 @@ void clear()
     }
 
     @Override
-    public Type getType()
+    public EventType getEventType()
     {
         return type;
     }
 
     @Override
+    public Type getType() {
+        if (type instanceof Type)
```

Comment:
I'd be more inclined to drop the Reactor events from the Type given that there are currently no users of those events yet (its never been released), whereas the others have been part of the main API for using the engine for quite some time and changing them would definitely hurt a bunch of folks. That said, my comment wasn't really about anything like that, just noting a dislike of needing to do the instanceof but also recognising why :)

I'm not sure about using 'Egnine'. I did actually think of that name earlier and decided agaisnt suggesting it. I'd say the older events are 'more core' in some ways than the new reactor ones, with the reactor being built on top of the eingine...but I guess also has IO related events that dont. Having both 'engine' and 'reactor' events named Core right now seems nicer in some ways than having the reactor events named 'Engine' events. I dunno. Mainly I just disliked 'Builtin' :)